mtags/src/main/scala-3/scala/meta/internal/pc/completions/Completions.scala

```diff
@@ -65,6 +67,77 @@ class Completions(
     (values, result)
   end completions
 
+  private def toCompletionValues(
+      completion: Completion
+  ): List[CompletionValue.Compiler] =
+    val shouldAddSnippet = path match
+      // if the parent is an apply then we already have ()
+      case (fun) :: (appl: GenericApply) :: _ if appl.fun == fun =>
+        false
+      case (_: Import) :: _ => false
+      case _ :: (_: Import) :: _ => false
+      case _ => true
+
+    completion.symbols.flatMap(
+      completionsWithSuffix(_, completion.label, shouldAddSnippet)
+    )
+  end toCompletionValues
+
+  inline private def undoBacktick(label: String): String =
+    label.stripPrefix("`").stripSuffix("`")
+
+  private def findApplySymbols(denot: Denotation): List[Symbol] =
+    denot match
+      case MultiDenotation(denot1, denot2) =>
+        List(findApplySymbols(denot1), findApplySymbols(denot2)).flatten
+      case NoDenotation => Nil
+      case _ =>
+        List(denot.symbol)
+
+  private def findSuffix(
+      shouldAddSnippet: Boolean,
+      methodSymbol: Symbol
+  ): Option[String] =
+    if shouldAddSnippet &&
+      config.isCompletionSnippetsEnabled &&
+      methodSymbol.is(Flags.Method)
+    then
+      val paramss = methodSymbol.paramSymss
+      paramss match
+        case Nil => None
+        case List(Nil) => Some("()")
+        case _ =>
+          val onlyImplicitOrTypeParams = paramss.forall(
+            _.exists { sym =>
+              sym.isType || sym.is(Implicit) || sym.is(Given)
+            }
+          )
+          if onlyImplicitOrTypeParams then None
+          else Some("($0)")
```

Comment:
I'm wondering can we make something like this 

```suggestion
          else if shouldAddSnippet && config.isCompletionSnippetsEnabled then Some("($0)")
          else Some("()")
```

and the top if predicate makes 

```diff
-    if shouldAddSnippet &&
-      config.isCompletionSnippetsEnabled &&
-      methodSymbol.is(Flags.Method)
+      methodSymbol.is(Flags.Method)
```

so that Metals can add `()` for methods defined with empty parens even if clients support snippet.

Reply:
Changed! Good idea! That's actually in line with the Scala 2 behaviour.